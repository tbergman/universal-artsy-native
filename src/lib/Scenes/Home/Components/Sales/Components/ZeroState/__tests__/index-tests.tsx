import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"
import { ZeroState } from "../index"

jest.mock("WebView", () => "WebView")
jest.mock("../index.html", () => "")

it("looks correct when rendered", () => {
  const auctions = renderer.create(<ZeroState />)
  expect(auctions).toMatchSnapshot()
})

describe("webview redirects", () => {
  it("should take over clicks", () => {
    const z = new ZeroState(null)
    const clickEvent: any = {
      navigationType: "click",
    }
    expect(z.shouldLoadRequest(clickEvent)).toBeFalsy()
  })

  it("should ignore redirects etc", () => {
    const z = new ZeroState(null)
    const redirectEvent: any = {
      navigationType: "redirect",
    }
    expect(z.shouldLoadRequest(redirectEvent)).toBeTruthy()
  })
})
