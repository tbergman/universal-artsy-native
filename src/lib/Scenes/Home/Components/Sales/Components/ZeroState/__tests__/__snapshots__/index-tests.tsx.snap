// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks correct when rendered 1`] = `
<WebView
  onShouldStartLoadWithRequest={[Function]}
  source=""
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
